### ISSUE TITLE
Support for data indexes for efficient search of jobs

### ISSUE THREAD

**Issue by manast** [taskforcesh/bullmq-pro-support#8]
Sometimes it is necessary to search for a job in the queue based on some data field. This is currently possible but it can only be done by using a linear search which can be really slow if there are a lot of jobs.

By supporting indexes natively for the jobs in the queue, every time a job is added/removed from the queue, the indexes will be updated. When doing a search, the indexes will be used if possible in order to accelerate such a search.

**Comment by StarpTech** [comment]
Quick question: This is currently on backlog for more than 2y. We are considering buying BullMQ Pro. Is the project still actively maintained or is GitHub Projects just outdated and how can we follow the roadmap? I came from the [docs](https://docs.bullmq.io/bullmq-pro/introduction) link. Thank you.

**Comment by manast** [comment]
Yes, it is maintained regularly, but this particular feature has not gained so much priority.
Here you can see the changelog: https://docs.bullmq.io/bullmq-pro/changelog
Hopefully as we are extending the dev team we can improve the communication of the roadmap in a much transparent way.

**Comment by StarpTech** [comment]
Thank you.
